System/Vector.cpp

```diff
@@ -53,32 +53,14 @@ namespace RTE {
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 	float Vector::GetAbsRadAngle() const {
-		if (m_X == 0) {
-			return (m_Y > 0) ? -c_HalfPI : ((m_Y < 0) ? c_HalfPI : 0);
-		}
-		if (m_Y == 0) {
-			return (m_X > 0) ? 0 : ((m_X < 0) ? c_PI : 0);
-		}
-		// TODO: Confirm that this is correct!")
-		float rawAngle = -std::atan(m_Y / m_X);
-		if (m_X < 0) { rawAngle += c_PI; }
-
-		return rawAngle;
+		float radAngle = -std::atan2(m_Y,m_X);
+		return (radAngle < -c_HalfPI) ? (radAngle + c_TwoPI) : radAngle;
 	}
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 	float Vector::GetAbsDegAngle() const {
-		if (m_X == 0) {
-			return (m_Y > 0) ? -90 : ((m_Y < 0) ? 90 : 0);
-		}
-		if (m_Y == 0) {
-			return (m_X > 0) ? 0 : ((m_X < 0) ? 180 : 0);
-		}
-		float rawAngle = -(std::atan(m_Y / m_X) / c_PI) * 180;
-		if (m_X < 0) { rawAngle += 180; }
-
-		return rawAngle;
+		return GetAbsRadAngle() / c_PI * 180.0F;
```

Comment:
Cleaning up all these things to reuse code is great! If it's doable, probably worth moving some of these one line definitions to the header.